packages/contracts-bedrock/scripts/Deployer.sol

```diff
@@ -54,60 +72,142 @@ abstract contract Deployer is Script, Artifacts {
         );
     }
 
+    /// @notice Call this function to sync the deployment artifacts such that
+    ///         hardhat deploy style artifacts are created.
+    function sync() public {
+        Deployment[] memory deployments = _getTempDeployments();
+        console.log("Syncing %s deployments", deployments.length);
+        console.log("Using deployment artifact %s", deployPath);
+
+        for (uint256 i; i < deployments.length; i++) {
+            address addr = deployments[i].addr;
+            string memory deploymentName = deployments[i].name;
+
+            string memory deployTx = _getDeployTransactionByContractAddress(addr);
+            if (bytes(deployTx).length == 0) {
+                console.log("Deploy Tx not found for %s skipping deployment artifact generation", deploymentName);
+                continue;
+            }
+            string memory contractName = _getContractNameFromDeployTransaction(deployTx);
+            console.log("Syncing deployment %s: contract %s", deploymentName, contractName);
+
+            string[] memory args = getDeployTransactionConstructorArguments(deployTx);
+            bytes memory code = _getCode(contractName);
+            bytes memory deployedCode = _getDeployedCode(contractName);
+            string memory receipt = _getDeployReceiptByContractAddress(addr);
+
+            string memory artifactPath = string.concat(deploymentsDir, "/", deploymentName, ".json");
+
+            uint256 numDeployments = 0;
+            try vm.readFile(artifactPath) returns (string memory res) {
+                numDeployments = stdJson.readUint(string(res), "$.numDeployments");
+                vm.removeFile(artifactPath);
+            } catch { }
+            numDeployments++;
+
```

Comment:
I had to [revert a commit](https://github.com/polymerdao/optimism-dev/pull/4/commits/537de6675bc6e3f2d270e278c797401e9547331e) to be able to deploy smart contracts using this branch. If the upstream updated their docs such that we can deploy OP contracts without a problem, then we shouldn't include these changes.